src/Kobo/Kernel.cpp

```diff
@@ -123,3 +128,16 @@ try {
 } catch (const std::runtime_error &e) {
   return false;
 }
+
+bool
+IsKoboOTGHostMode()
+try {
+#ifdef KOBO
+  if (DetectKoboModel() != KoboModel::CLARA_HD) return IsKoboCustomKernel();
+  /* for Clara HD, just test for existence of host mode file */
+  FileReader file(Path("/mnt/onboard/XCSoarData/kobo/OTG_Host_Active"));
```

Comment:
This is a clumsy way of testing a file's existence; better use `File::Exists()`.